file-compiler-plugin/src/main/java/io/ballerina/stdlib/file/compiler/FileServiceValidator.java

```diff
@@ -0,0 +1,128 @@
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.stdlib.file.compiler;
+
+import io.ballerina.compiler.syntax.tree.FunctionDefinitionNode;
+import io.ballerina.compiler.syntax.tree.FunctionSignatureNode;
+import io.ballerina.compiler.syntax.tree.ParameterNode;
+import io.ballerina.compiler.syntax.tree.RequiredParameterNode;
+import io.ballerina.compiler.syntax.tree.SeparatedNodeList;
+import io.ballerina.compiler.syntax.tree.ServiceDeclarationNode;
+import io.ballerina.compiler.syntax.tree.SyntaxKind;
+import io.ballerina.projects.plugins.AnalysisTask;
+import io.ballerina.projects.plugins.SyntaxNodeAnalysisContext;
+import io.ballerina.tools.diagnostics.Diagnostic;
+import io.ballerina.tools.diagnostics.DiagnosticFactory;
+import io.ballerina.tools.diagnostics.DiagnosticInfo;
+import io.ballerina.tools.diagnostics.DiagnosticSeverity;
+import io.ballerina.tools.diagnostics.Location;
+import org.ballerinalang.util.diagnostic.DiagnosticErrorCode;
+
+/**
+ * File service validator for compiler API.
+ */
+public class FileServiceValidator implements AnalysisTask<SyntaxNodeAnalysisContext> {
+    private static final String CODE = "File";
+    private static final String FILE_EVENT = "file:FileEvent";
+    private static final String RESOURCE_NAME_ON_CREATE = "onCreate";
+    private static final String RESOURCE_NAME_ON_DELETE = "onDelete";
+    private static final String RESOURCE_NAME_ON_MODIFY = "onModify";
+    private static final String INVALID_INPUT_PARAM = "Invalid parameter type `{0}` provided for remote function. " +
+            "Only file:FileEvent is allowed as the parameter type";
+    private static final String INVALID_REMOTE_FUNCTION = "Missing remote key word in the remote function `{0}`";
+    private static final String INVALID_FUNCTION_NAME = "Invalid function name `{0}`";
+    private static final String INVALID_RETURN_TYPE = "Return types are not allowed in the remote function `{0}`";
+    private static final String INVALID_PARAM_SIZE = "The remote function should contain a parameter";
+    private static final String EMPTY_SERVICE = "At least a single remote function required in the service";
+
+    @Override
+    public void perform(SyntaxNodeAnalysisContext syntaxNodeAnalysisContext) {
+        ServiceDeclarationNode serviceDeclarationNode = (ServiceDeclarationNode) syntaxNodeAnalysisContext.node();
+        long size = serviceDeclarationNode.members().stream().filter(child -> child.kind() ==
```

Comment:
Here we need to filter out only the file package-related services. 

You can refer to this commit https://github.com/ballerina-platform/module-ballerina-grpc/pull/158/commits/fecfeb3bf4fbfd10a95fa8ddc7442962ed172c01

Reply:
Better if we can implement a test case to capture this use case, I mean a test case with multiple services including file listener.

Reply:
Fixed in 4d82650